actionpack/lib/action_controller/renderer.rb

```diff
@@ -72,8 +72,19 @@ def with_defaults(defaults)
     def initialize(controller, env, defaults)
       @controller = controller
       @defaults = defaults
-      @env = normalize_keys defaults, env
+      if env.blank? && @defaults == DEFAULTS
+        @env = DEFAULT_ENV
+      else
+        @env = normalize_env(@defaults)
+        @env.merge!(normalize_env(env)) unless env.blank?
+      end
+    end
+
+    def defaults # :nodoc:
```

Comment:
Why is this nodoc and then add the `attr_reader`? It is fine to change from a reader to a method in the API.